internal/core/application/mocks_test.go

```diff
@@ -1,12 +1,11 @@
-package application_test
+package application
```

Comment:
nACK.

> This updates application tests pkg name as packages should have same name as file/func that is tested.

Tests **must not** have the same package name of the tested one. The usefulness of `_test` suffix is:
1. tests are not imported in case the tested package is required by another one.
2. as being in a different package, the tests have the same view of the tested package as those actually importing it for concrete use. This way only exported types and function can be tested since they will be the only usable.

Please revert the change.

Reply:
I see there are people in community that are using both "white-box" and "black-box" testing techniques, one such discussion can be found in [here](https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language).
"Black box" technique is used, as u said, to prevent circular dependencies and to test only exported stuff.
We should not have issues with circular dependencies as our layers are divided vertically, meaning application can import lower domain level and not vice verse, and imported pkg's do not import any of our pkg's.
Other reason, "to test only exported stuff", seems to me as disadvantage in out case since we want to do unit testing and test both exported and unexported funcs like in my case initGroupedVolume. 
I see that in "black box" testing it is recommended to use additional file with suffix *_internal_test.go which would have same pkg name as unexported func being tests.

Anyways, my primary motivation, for updating test pkg name, was to test initGroupedVolume func, so how do you want to proceed with it? to leave changes i did (update test pkg name) or to create additional test file for this fucn with suffix *_internal_test.go ? @tiero @altafan

Reply:
If we are renaming current test suites for this only function, I would go with `*_internal_test.go` approach

Reply:
I understand that both "white" and "black" box testing have their pros and cons, but in general, I think this change is something that requires a proper flow to be eventually committed, therefore opening an issue, and eventually proceeding with changes AFTER a discussion.

I don't like the idea of this PR of having the application layer white-box tested, while the rest of the tests are ALL black-box. If the change is done, it must be done for ALL packages.

Another thing I don't like so much right now is having utils function sparse in the application layer. In fact, You can see `repo_utils.go` and `explorer_utils.go` as attempts  of grouping them into one or more util files.

My proposal is: what if we pack util functions altogether (or divided in more `xxx_util.go` in a way that makes sense)  and we "white-box" test only this/these files, leaving the rest "black-box" as already done?

Reply:
> I understand that both "white" and "black" box testing have their pros and cons, but in general, I think this change is something that requires a proper flow to be eventually committed, therefore opening an issue, and eventually proceeding with changes AFTER a discussion.

I don't like the idea of this PR of having the application layer white-box tested, while the rest of the tests are ALL black-box. If the change is done, it must be done for ALL packages.

Agree.

> Another thing I don't like so much right now is having utils function sparse in the application layer. 

Dont agree with this one, from my pov it is ok to have private methods/functions inside application service files, this encourages splitting complex service methods to more granular funcs which improves testability and readability.
Utils methods should be like pkg's, something could be reusable in various places.